### DISCUSSION TITLE
.NET 6/minimal APIs

### DISCUSSION THREAD

**Discussion by rich-mond** [pdevito3/craftsman#44] [Ideas]
Hi @pdevito3 

Just curious to get your take on minimal APIs that are part of the up and coming .NET 6 release, to be integrated into craftsman in the future? 

Good reference here: https://gist.github.com/davidfowl/ff1addd02d239d2d26f4648a06158727

As the craftsman generated controllers are pretty light, theres potential to gain in simplified API definitions via the new minimal approach. With less overhead there are performance gains as well.

Another useful feature of C# 10 is global usings that simplify code generation by centralising the definitions.

**Comment by pdevito3** [comment]
yup, this is a good question. 

the short answer is i probably won't do anything with these, but let me elaborate a bit:

regarding the minimal apis, i like them, but they don't support everything that's in use, especially swagger comments. i think there might be swagger attributes that would work here, but from what i've seen minimal api's aren't meant to replace controllers, just provide a more minimal option when it makes sense. not sure that's the case here.

the global usings are an interesting one. they seem like they could be nice to use in some cases, but might cause clashing issues in some cases so i'd want to rely on it sparingly

**Reply by rich-mond** [reply]
Makes sense 👍
